app/components/primer/alpha/action_menu/action_menu_element.ts

```diff
@@ -0,0 +1,135 @@
+import '@github/include-fragment-element'
+
+const popoverSelector = (() => {
+  try {
+    document.querySelector(':open')
+    return ':open'
+  } catch {
+    return '.\\:open'
+  }
+})()
+
+type SelectVariant = 'single' | 'multiple' | null
+
+const menuItemSelectors = ['[role="menuitem"]', '[role="menuitemcheckbox"]', '[role="menuitemradio"]']
+
+export class ActionMenuElement extends HTMLElement {
+  #abortController: AbortController
+  #originalLabel = ''
+
+  get selectVariant(): SelectVariant {
+    return this.getAttribute('data-select-variant') as SelectVariant
+  }
+
+  set selectVariant(variant: SelectVariant) {
+    if (variant) {
+      this.setAttribute('data-select-variant', variant)
+    } else {
+      this.removeAttribute('variant')
+    }
+  }
+
+  get dynamicLabelPrefix(): string {
+    const prefix = this.getAttribute('data-dynamic-label-prefix')
+    if (!prefix) return ''
+    return `${prefix}:`
+  }
+
+  set dynamicLabelPrefix(value: string) {
+    this.setAttribute('data-dynamic-label', value)
+  }
+
+  get dynamicLabel(): boolean {
+    return this.hasAttribute('data-dynamic-label')
+  }
+
+  set dynamicLabel(value: boolean) {
+    this.toggleAttribute('data-dynamic-label', value)
+  }
+
+  get popoverElement(): HTMLElement | null {
+    return this.querySelector<HTMLElement>('[popover]')
+  }
+
+  get invokerElement(): HTMLElement | null {
+    const id = this.querySelector('[role=menu]')?.id
```

Comment:
Right now, we have two `role="menu"` declarations. One on the custom element, and another on the `ul`. I _think_ it should only be on the `ul` as its the direct parent of the items, but I could be wrong. It just shouldn't be in both places (that I know for sure)

Reply:
`Overlay` (`<anchored-position popover>`) requires either `role=dialog` or `role=menu`. `ActionMenu::List` (`<ul>`) enforces `role=menu`. We can make Overlay's role use optional or omit the enforcement in `ActionMenu::List`.